Events / customer relationship updates

raldea opened the pull request.

raldea committed:
Created new system config, transferred form in custom my account page, added customer group restriction

raldea committed:
Created events list and storefront even edit, created relationship between event and customer assignable in storefront and admin

jakwinkler commented:
@raldea this is a great PR and I would love to merge it, however at @qoliber we follow these rules:
- `FQDN` for variable types in `PHPDoc`


The field:
```xml
 <field name="customer_id"
               component="Qoliber_EventCalendar/js/form/element/formulary/auto-complete"
               sortOrder="500" formElement="input">
            <argument name="data" xsi:type="array">
                <item name="config" xsi:type="array">
                    <item name="source" xsi:type="string">customer</item>
                    <item name="url" xsi:type="url" path="events/event/customerList"/>
                    <item name="noticeModal" xsi:type="string" translate="true"><![CDATA[ If you proceed,
                     the customer will be unlinked from the current event and linked to the selected customer.]]>
                    </item>
                    <item name="noticeTitle" xsi:type="string" transla
```

needs to be changed, it should not remove the customer_id when the event is saved / udpated in admin interface.
(I might read the code wrong, barely awake ;-))
I'll test the code in 24.0h from now (30-07-2025)
Please take a look how it is done in the Case Study module here:

https://github.com/qoliber/m2-case-study-manager/blob/main/view/adminhtml/ui_component/qoliber_casestudy_form.xml

```xml
<field name="customer_id" component="Magento_Ui/js/form/element/ui-select" sortOrder="20" formElement="select">
            <argument name="data" xsi:type="array">
                <item name="config" xsi:type="array">
                    <item name="source" xsi:type="string">casestudy</item>
                    <item name="filterOptions" xsi:type="boolean">true</item>
                    <item name="multiple" xsi:type="boolean">false</item>
                    <item name="showCheckbox" xsi:type="boolean">false</item>
                    <item name="disableLabel" xsi:type="boolean">true</item>
                    <item name="isRemoveSelectedIcon" xsi:type="boolean">false</item>
                    <item name="levelsVisibility" xsi:type="number">1</item>
                </item>
            </argument>
            <settings>
                <elementTmpl>ui/grid/filters/elements/ui-select</elementTmpl>
                <label translate="true">Customer</label>
                <dataScope>customer_id</dataScope>
                <componentType>field</componentType>
                <validation>
                    <rule name="required-entry" xsi:type="boolean">true</rule>
                </validation>
            </settings>
            <formElements>
                <select>
                    <settings>
                        <options class="Qoliber\CaseStudyManager\Model\Config\Source\Customer"/>
                    </settings>
                </select>
            </formElements>
        </field>
```

raldea committed:
Qoliber/EventCalendar updated autocomplete search to select in events form admin

raldea commented:
Thanks for the feedback @jakwinkler.

I  updated the PR:
- Changed customer selection from autosearch to dropdown select with search
- Removed files related to autosearch component
- Retested overall changes and confirmed still working as expected

See commit: 
https://github.com/qoliber/m2-event-calendar/pull/8/commits/c1eef8357b9481b89cace390aeb99d3300c27197

jakwinkler commented:
You sure it will work now? :-) 

```xml
            <argument name="data" xsi:type="array">
                <item name="config" xsi:type="array">
                    <item name="source" xsi:type="string">casestudy</item>
                    <item name="filterOptions" xsi:type="boolean">true</item>
                    <item name="multiple" xsi:type="boolean">false</item>
                    <item name="showCheckbox" xsi:type="boolean">false</item>
                    <item name="disableLabel" xsi:type="boolean">true</item>
                    <item name="isRemoveSelectedIcon" xsi:type="boolean">false</item>
                    <item name="levelsVisibility" xsi:type="number">1</item>
                </item>
            </argument>

```

refers to casestyudy now :-) the source is different
also, if you want me to merge the code, I do require FQDN in PHPDOcs

raldea committed:
Qoliber/EventCalendar updated ui component source for customer select, applied FQDN in PHPDocs

raldea committed:
Qoliber/EventCalendar updated ui component source for customer select, applied FQDN in PHPDocs

raldea commented:
Hi @jakwinkler.

- Confirmed all items in the select ui component are needed to render a single select with search. Except for the source, which is now updated
- Also updated PHPDocs to follow FQDN
- Tested both admin and storefront submit and confirmed working

LMK if you have any questions

jakwinkler commented:
Released in `1.1.0`
